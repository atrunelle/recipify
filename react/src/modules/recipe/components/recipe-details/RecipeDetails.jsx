import React, { Component } from 'react';
import { Card, CardContent, Typography } from '@material-ui/core';

class RecipeDetails extends Component {
  render() {
    if (!this.props.recipe) return false;
    
    const ingredientsList = this.props.recipe.ingredients.map((ingredient, index) => 
      <li key={index}>{ingredient.name}</li>
    );
    
    return (
      <Card>
        <CardContent>
          <Typography
            gutterBottom
            variant="headline"
            component="h2">
            {this.props.recipe.name}
          </Typography>
          <ul>
            {ingredientsList}
          </ul>
        </CardContent>
      </Card>
    );
  }
}

export default RecipeDetails;