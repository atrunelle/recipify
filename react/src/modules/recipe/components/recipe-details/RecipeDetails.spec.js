import React from 'react';
import { shallow } from 'enzyme';
import RecipeDetails from './RecipeDetails';

describe('Component: RecipeDetails', () => {
  it('should not render if no recipe is passed as prop', () => {
    const component = shallow(<RecipeDetails />);
    expect(component.html()).toBeNull();
  });

  it('should match snapshot', () => {
    const mockRecipe = {
      name: 'My recipe',
      ingredients: [{
        name: 'carrot',
      }, {
        name: 'spinach',
      }],
    };
    const component = shallow(<RecipeDetails recipe={mockRecipe}/>);
    expect(component.html()).toMatchSnapshot();
  });
});