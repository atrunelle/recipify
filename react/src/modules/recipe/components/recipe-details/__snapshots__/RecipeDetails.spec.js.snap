// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RecipeDetails should match snapshot 1`] = `"<div class=\\"MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 MuiCard-root-1\\"><div class=\\"MuiCardContent-root-29\\"><h2 class=\\"MuiTypography-root-30 MuiTypography-headline-35 MuiTypography-gutterBottom-47\\">My recipe</h2><ul><li>carrot</li><li>spinach</li></ul></div></div>"`;
